sample/migrate_rt.py

```diff
@@ -0,0 +1,75 @@
+"""
+The configuration file would look like this:
+
+{
+    "authority": "https://login.microsoftonline.com/organizations",
+    "client_id": "your_client_id",
+    "scope": ["User.ReadBasic.All"],
+        // You can find the other permission names from this document
+        // https://docs.microsoft.com/en-us/graph/permissions-reference
+}
+
+You can then run this sample with a JSON configuration file:
+
+    python sample.py parameters.json
+"""
+
+import sys  # For simplicity, we'll read config file from 1st CLI param sys.argv[1]
+import json
+import logging
+
+import msal
+
+
+# Optional logging
+# logging.basicConfig(level=logging.DEBUG)  # Enable DEBUG log for entire script
+# logging.getLogger("msal").setLevel(logging.INFO)  # Optionally disable MSAL DEBUG logs
+
+config = json.load(open(sys.argv[1]))
+
+def get_rt_via_old_app():
+    # Let's pretend this is an old app powered by ADAL
```

Comment:
Suggestion: I think we should  use ADAL here. It is confusing to understand when the comment mentions that this is an app powered by ADAL but is using MSAL in the very next line.

Reply:
+1 to use this as a way to demonstrate a typical scenario and explain also the resource to scope fun

---
In reply to: [424830506](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/193#discussion_r424830506) [](ancestors = 424830506)

Reply:
Interestingly, this script was created as a [quick-and-dirty test driver](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/193/files#diff-615899d8b1282d9f785b5404cf87e0ccR48-R52), and it is already unfortunate that it ended up with much more lines in preparing an old RT than demonstrating the actual migration (thus distracting a sample reader). AND now it does attract more attentions and comments than necessary.

Here is the thing.

* The old app does not have to be an ADAL Python app. It could also be an app developer wanting to migrate his/her app powered by a 3rd-party generic OAuth2 library to MSAL.
* Even if the old app is using ADAL Python, there is actually not a specific thing as "adal python token cache". ADAL Python never enforced a particular token cache persistence format. To test such migration, we would need to call an old (adal?) app, acquire tokens, INVENT an ad-hoc token storage format, pretend to read token from it, ..., etc.. Our ad-hoc token storage format does not necessarily match what an ADAL Python app developer's situation, thus might just be noise to them.

Now I incline to replace this entire `def get_rt_via_old_app()` implementation into something like this:

```python
def get_rt_via_old_app():
    # Maybe you have an ADAL-powered app like this
    # https://github.com/AzureAD/azure-activedirectory-library-for-python/blob/dev/sample/device_code_sample.py
    # or you have an app already talking to Microsoft identity platform v2,
    # powered by some 3rd-party auth library.
    # And you already persist its tokens somehow.
    # You are expected to extract RTs from there, and return it like this.
    return "old_rt", ["scope1", "scope2"]
```

which is very close to the [previous doc here](https://docs.microsoft.com/en-us/azure/active-directory/develop/migrate-python-adal-msal#migrate-existing-refresh-tokens-for-msal-python).

Reply:
Yes, makes sense. However here, we are still skipping the part where we explain what scope1 and scope2 mean and if they are still using resources to acquire this RT what would that conversion from resource to scope look like.

Reply:
See the latest commit.